vowpalwabbit/core/src/reductions/explore_eval.cc

```diff
@@ -184,6 +198,10 @@ void do_actual_learning(explore_eval& data, multi_learner& base, VW::multi_ex& e
 
     if (data.random_state->get_and_update_random() < threshold)
     {
+      // if we reached the example quota for this example block just skip
+      if (data.block_ex_quota == 0) { return; }
```

Comment:
isn't it more readable to have it as elseif on line 172?

Reply:
yeah probably, updated

Reply:
ah no if we skip too early then know_cost isn't changed